test/jest/config.js

```diff
@@ -0,0 +1,26 @@
+const path = require('path');
+
+module.exports = {
+  collectCoverageFrom: [
+    'ui_framework/components/**/*.js',
+    // Seems to be a bug with jest or micromatch, in which the above glob
+    // doesn't match subsequent levels of directories, making this glob necessary.
+    'ui_framework/components/**/**/*.js',
+    '!ui_framework/components/index.js',
+    '!ui_framework/components/**/*/index.js',
+  ],
+  coverageDirectory: '<rootDir>/target/test_coverage',
```

Comment:
I like having one directory for everything that's generated (normally in the Java world that's `target`). Just so used to that from other projects. But I'll move it to follow the "Kibana way".